json-rpc/src/views.rs

```diff
@@ -60,47 +60,55 @@ pub struct EventView {
 #[derive(Debug, Serialize, Deserialize)]
 #[serde(tag = "type")]
 pub enum EventDataView {
+    #[serde(rename = "receivedpayment")]
     ReceivedPayment {
         amount: u64,
         sender: BytesView,
         metadata: BytesView,
     },
+    #[serde(rename = "sentpayment")]
     SentPayment {
         amount: u64,
         receiver: BytesView,
         metadata: BytesView,
     },
-    SystemEvent,
+    #[serde(rename = "unknown")]
+    Unknown {},
 }
 
-impl TryFrom<(u64, ContractEvent)> for EventView {
-    type Error = Error;
-
+impl From<(u64, ContractEvent)> for EventView {
     /// Tries to convert the provided byte array into Event Key.
-    fn try_from((txn_version, event): (u64, ContractEvent)) -> Result<EventView, Error> {
+    fn from((txn_version, event): (u64, ContractEvent)) -> EventView {
         let event_data = if event.type_tag() == &TypeTag::Struct(received_payment_tag()) {
-            let received_event = ReceivedPaymentEvent::try_from(&event)?;
-            EventDataView::ReceivedPayment {
-                amount: received_event.amount(),
-                sender: BytesView::from(received_event.sender().as_ref()),
-                metadata: BytesView::from(received_event.metadata()),
+            if let Ok(received_event) = ReceivedPaymentEvent::try_from(&event) {
+                Ok(EventDataView::ReceivedPayment {
+                    amount: received_event.amount(),
+                    sender: BytesView::from(received_event.sender().as_ref()),
+                    metadata: BytesView::from(received_event.metadata()),
+                })
+            } else {
+                Err(format_err!("Unable to parse ReceivedPaymentEvent"))
             }
         } else if event.type_tag() == &TypeTag::Struct(sent_payment_tag()) {
-            let sent_event = SentPaymentEvent::try_from(&event)?;
-            EventDataView::SentPayment {
-                amount: sent_event.amount(),
-                receiver: BytesView::from(sent_event.receiver().as_ref()),
-                metadata: BytesView::from(sent_event.metadata()),
+            if let Ok(sent_event) = SentPaymentEvent::try_from(&event) {
+                Ok(EventDataView::SentPayment {
+                    amount: sent_event.amount(),
+                    receiver: BytesView::from(sent_event.receiver().as_ref()),
+                    metadata: BytesView::from(sent_event.metadata()),
+                })
+            } else {
+                Err(format_err!("Unable to parse ReceivedPaymentEvent"))
             }
         } else {
-            EventDataView::SystemEvent
+            Err(format_err!("Unknown events"))
```

Comment:
maybe I misread that, but it's not ok to do, right?
if smart contract emits some unknown event(e.g. reconfiguration event) then this conversion will return error, and then rpc method will bubble this error up. 
so one system event will break whole API call

Reply:
No, 
on line 110: 
 data: event_data.unwrap_or(EventDataView::Unknown {}), 

This will convert any errors encountered during EventDataView conversion into an Unknown {} 

So it will never fail.

Reply:
Just wondering: should we add a `message` field in `UnknownEvents` then to store the error? Seems like all the different errors we set when trying to parse event above just gets swallowed into the same nondescript `EventDataView::Unknown`